intl.emu

```diff
@@ -142,57 +138,75 @@
       <dl class="header">
         <dt>description</dt>
         <dd>
-          <p><ins>If _x_ is a mathematical value, it</ins> <del>It</del> involves solving the following equation, which returns a valid mathematical value given integer inputs:</p>
+          <p>It involves solving the following equation, which returns a valid mathematical value given integer inputs:</p>
           <emu-eqn id="eqn-ToRawPrecisionFn" aoid="ToRawPrecisionFn">
             ToRawPrecisionFn(_n_, _e_, _p_) = _n_ × 10<sup>_e_–_p_+1</sup>
             where 10<sup>_p_–1</sup> ≤ _n_ < 10<sup>_p_</sup>
           </emu-eqn>
         </dd>
       </dl>
       <emu-alg>
-        1. <ins>If _x_ is a Decimal128 value, then</ins>
-          1. <ins>Let _m_ be Decimal128ToDecimalString(_x_, *false*).</ins>
-          1. <ins>Let _e_ be the truncated exponent of _x_.</ins>
-          1. <ins>Let _n_ be the scaled significand of _x_.</ins>
-        1. <ins>Else,</ins>
-          1. <ins>Assert _x_ is a mathematival value.</ins>
-          1. Let _p_ be _maxPrecision_.
-          1. If _x_ = 0<ins> or _x_ is a zero Decimal128 object</ins>, then
-            1. Let _m_ be the String consisting of _p_ occurrences of the code unit 0x0030 (DIGIT ZERO).
-            1. Let _e_ be 0.
-            1. Let _xFinal_ be 0.
+        1. <ins>If _x_ is a Decimal128 value, return Decimal128ToRawPrecision(_x_, _minPrecision_, _maxPrecision_, _unsignedRoundingMode_).</ins>
+        1. Let _p_ be _maxPrecision_.
```

Comment:
I haven't verified, but I assume `<ins>If _x_ is a Decimal128 value, return Decimal128ToRawPrecision(_x_, _minPrecision_, _maxPrecision_, _unsignedRoundingMode_).</ins>` is the one line added, and everything else is being reset to what it was before. (That is the correct thing to do)

Reply:
I took another look and decided to not factor out a helper AO that works just for decimals. The rationale is that ToRawFixed and ToRawPrecision to the right thing; they just need a bit of pre-processing of a decimal argument before doing inspection & surgery on decimal strings.